lib/datadog/tracing/contrib/propagation/sql_comment.rb

```diff
@@ -12,36 +14,30 @@ module SqlComment
           def self.annotate!(span_op, mode)
             return unless mode.enabled?
 
-            # PENDING: Until `traceparent`` implementation in `full` mode
-            # span_op.set_tag(Ext::TAG_DBM_TRACE_INJECTED, true) if mode.full?
+            span_op.set_tag(Ext::TAG_DBM_TRACE_INJECTED, true) if mode.full?
           end
 
-          def self.prepend_comment(sql, span_op, mode)
+          def self.prepend_comment(sql, digest, mode)
             return sql unless mode.enabled?
 
             tags = {
-              Ext::KEY_DATABASE_SERVICE => span_op.service,
+              Ext::KEY_DATABASE_SERVICE => digest.span_service,
               Ext::KEY_ENVIRONMENT => datadog_configuration.env,
               Ext::KEY_PARENT_SERVICE => datadog_configuration.service,
               Ext::KEY_VERSION => datadog_configuration.version
             }
 
-            # PENDING: Until `traceparent`` implementation in `full` mode
-            # tags.merge!(trace_context(span_op)) if mode.full?
+            if mode.full?
```

Comment:
The `trace_op.to_digest` is expensive'ish to create.
It's totally ok to call it when propagation is necessary, but I recommend not creating a digest if we are not going to use it for propagation. In other words, if `mode.full?` is false, I think we shouldn't call `trace_op.to_digest`, as this will make all traces slower, even the ones that don't have SQL comment propagation enabled.